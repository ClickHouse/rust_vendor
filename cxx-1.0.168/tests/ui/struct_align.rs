#[cxx::bridge]
mod ffi {
    #[repr(align(3))]
    struct SharedA {
        b: [u8; 4],
    }

    // 1073741824 = 2^30
    #[repr(align(1073741824))]
    struct SharedB {
        b: [u8; 4],
    }

    #[repr(align(-2))]
    struct SharedC {
        b: [u8; 4],
    }
}

fn main() {}
