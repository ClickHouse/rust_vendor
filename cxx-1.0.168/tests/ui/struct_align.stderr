error: invalid repr(align) attribute: not a power of two
 --> tests/ui/struct_align.rs:3:18
  |
3 |     #[repr(align(3))]
  |                  ^

error: invalid repr(align) attribute: larger than 2^13
 --> tests/ui/struct_align.rs:9:18
  |
9 |     #[repr(align(1073741824))]
  |                  ^^^^^^^^^^

error: invalid repr(align) attribute: an arithmetic expression is not supported
  --> tests/ui/struct_align.rs:14:18
   |
14 |     #[repr(align(-2))]
   |                  ^^
